src/auth/mod.rs

```diff
@@ -0,0 +1,128 @@
+pub(super) use crate::actix_web::Result as ActixResult;
+use crate::actix_web::{error::ErrorUnauthorized, HttpRequest};
+use actix_web::http::header::HeaderMap;
+
+pub mod jwt;
+
+#[derive(Clone)]
+pub struct AuthHeader {
+    field: &'static str,
+    token_bound: (Option<&'static str>, Option<&'static str>),
+}
+
+impl AuthHeader {
+    /// return None if value is invalid or can't be parsed
+    pub fn new(field: &'static str, value: &'static str) -> Option<Self> {
+        let mut not_token = value.trim().split("{token}");
+        let token_bound = (
+            not_token.next().filter(|s| !s.is_empty()),
+            match not_token.next() {
+                None => return None,
+                Some(s) if s.is_empty() => None,
+                Some(s) => Some(s),
+            },
+        );
+        Some(Self { field, token_bound })
+    }
+}
+
+impl Default for AuthHeader {
+    fn default() -> Self {
+        AuthHeader::new("Authorization", "Bearer {token}").expect("has {token}")
```

Comment:
very clean in Rust ya, `.expect("has {token}")` amazing

Reply:
welcome to the Rust cultism :smile: